docs/project/principles/safety_strategy.md

```diff
@@ -1,771 +0,0 @@
-    so that C++ developers can readily take advantage of Carbon's improvements.
-
-    -   Safety mechanisms will ideally be designed to apply to automatically
-        migrated C++ code. Providing immediate safety improvements to Carbon
-        adopters will help motivate adoption.
-
-    -   In the other direction, safety mechanisms must not force manual
-        rewriting of C++ code in order to migrate, either by creating design
-        incompatibilities or performance degradations. Automated migration of
-        C++ code to Carbon must work for most developers, even if it forces
-        Carbon's safety design to take a different approach.
-
-    -   Carbon's safety should degrade gracefully when Carbon code calls C++
-        code, although this may require use of the Carbon toolchain to compile
-        the C++ code. Applications should be expected to use interoperability.
-        Although some safety features will be Carbon-specific, safety should not
-        stop at the language boundary.
-
--   The rules for determining whether code will pass compile-time safety
-    checking should be articulable, documented, and easy to understand.
-
-    -   Compile-time safety checks should not change significantly across
-        different build modes. The purpose of the build modes is to determine
-        code generation.
-
--   Each build mode will prioritize performance and safety differently:
-
-    -   The [debug build mode](#debug) will produce development-focused binaries
-        that prioritize fast iteration on code with safety checks that assist in
-        identification and debugging of errors.
-
-    -   The [performance build mode](#performance) will produce release-focused
-        binaries that prioritize performance over safety.
-
-    -   The [hardened build mode](#hardened) will produce release-focused
-        binaries that prioritize safety that is resistant to attacks at the cost
-        of performance.
-
--   Safety checks should try to be identical across build modes.
-
-    -   There will be differences, typically due to performance overhead and
-        detection rate trade-offs of safety check algorithms.
-
--   The number of build modes will be limited, and should be expected to remain
-    at the named three.
-
-    -   Most developers will use two build modes in their work: debug for
-        development, and either performance or hardened for releases.
-
-    -   It's important to focus on checks that are cheap enough to run as part
-        of normal development. Users are not expected to want to run additional
-        development build modes for additional sanitizers.
-
-    -   Limiting the number of build modes simplifies support for both Carbon
-        maintainers, who can focus on a more limited set of configurations, and
-        Carbon developers, who can easily choose which is better for their
-        use-case.
-
--   Each distinct safety-related build mode (debug, performance, and hardened)
```

Comment:
Is this still true? It would be good to have the answer to this documented, though it might be better in /docs/design/safety/README.md. I do think we expect "strict" files to be able to mixed with "permissive", which may also be worth saying.